### PR TITLE
Understanding OpenShift Pipelines

### PR DESCRIPTION
**Pull request by nkakkar81**
Added assembly for "understanding openshift pipelines"

### CODE REVIEW

**Comment by nkakkar81** [comment]
@Preeticp @boczkowska : Request you to please review this PR. I will send it for review to dev and QE, once you have reviewed. Thanks

**Review by vdemeester — APPROVED** [review]
/lgtm

File: openshift_pipelines/understanding-openshift-pipelines.adoc
Comment (siamaksade) [inline openshift_pipelines/understanding-openshift-pipelines.adoc:16]: I'd also add something along OpenShift Pipelines is a serverless CI/CD system that runs pipelines with all the dependencies they require in isolated containers.

File: modules/op-pipelines-concepts.adoc
Comment (siamaksade) [inline modules/op-pipelines-concepts.adoc:6]: Missing `Condition` and trigger objects

File: modules/op-pipelines-concepts.adoc
Comment (ppitonak) [inline modules/op-pipelines-concepts.adoc:18]: shouldn't it be `CloudEvent` (without space)?

File: modules/op-pipelines-concepts.adoc
Comment (nkakkar81) [reply modules/op-pipelines-concepts.adoc:14]: @ppitonak - The underscores make the word italics

File: openshift_pipelines/understanding-openshift-pipelines.adoc
Comment (nkakkar81) [reply openshift_pipelines/understanding-openshift-pipelines.adoc:17]: @siamaksade - Made changes, please verify. Thanks !!

File: modules/op-pipelines-concepts.adoc
Comment (nkakkar81) [reply modules/op-pipelines-concepts.adoc:6]: done, request you to kindly review. Thank you :)

File: modules/op-pipelines-concepts.adoc
Comment (nkakkar81) [reply modules/op-pipelines-concepts.adoc]: done

File: modules/op-pipelines-concepts.adoc
Comment (nkakkar81) [reply modules/op-pipelines-concepts.adoc]: done :)

File: modules/op-pipelines-concepts.adoc
Comment (ppitonak) [inline modules/op-pipelines-concepts.adoc:22]: "determine the sequence of Tasks in your Pipeline" sounds to me like it determines the order of tasks which IMHO isn't the main use.

File: modules/op-pipelines-concepts.adoc
Comment (ppitonak) [inline modules/op-pipelines-concepts.adoc:20]: `Triggers` is the only one in plural

File: modules/op-pipelines-concepts.adoc
Comment (nkakkar81) [reply modules/op-pipelines-concepts.adoc]: @ppitonak - Done with the changes, kindly review . Thanks :)

File: modules/op-pipelines-concepts.adoc
Comment (nkakkar81) [reply modules/op-pipelines-concepts.adoc]: removed s

**Review by ppitonak — APPROVED** [review]

**Comment by ppitonak** [comment]
LGTM

File: openshift_pipelines/understanding-openshift-pipelines.adoc
Comment (Preeticp) [inline openshift_pipelines/understanding-openshift-pipelines.adoc:20]: Not a part of your changes, but can you please remove 'project', both namespaces and project almost mean the same thing.
Under/in?

File: modules/op-pipelines-concepts.adoc
Comment (Preeticp) [inline modules/op-pipelines-concepts.adoc:9]: I know we have expanded CRD once in the assembly, but since we aim to enable each module to stand on its own, it probably makes sense to expand it the first time we use it in the module here. WDYT?

File: modules/op-pipelines-concepts.adoc
Comment (Preeticp) [inline modules/op-pipelines-concepts.adoc:14]: A PipelineRun initiates

File: modules/op-pipelines-concepts.adoc
Comment (Preeticp) [inline modules/op-pipelines-concepts.adoc:16]: It is the result of

File: modules/op-pipelines-concepts.adoc
Comment (Preeticp) [inline modules/op-pipelines-concepts.adoc:18]: ...Pipeline Tasks.

File: modules/op-pipelines-concepts.adoc
Comment (Preeticp) [inline modules/op-pipelines-concepts.adoc:20]: ...such as a Git...
remove comma after request
s/received/receives
Add comma after 'receives'
create a full-fledged CI/CD systems - remove 'a'
How about: You can use Triggers along with Pipelines, to create full-fledged CI/CD systems...?

File: modules/op-pipelines-concepts.adoc
Comment (Preeticp) [inline modules/op-pipelines-concepts.adoc:22]: check, which
fails/fail
all the subsequent
You can use Conditions in your Pipeline to create...?

**Comment by Preeticp** [comment]
Overall, looks great. Some nits and suggestions.

File: openshift_pipelines/understanding-openshift-pipelines.adoc
Comment (nkakkar81) [reply openshift_pipelines/understanding-openshift-pipelines.adoc]: @siamaksade ^^

File: modules/op-pipelines-concepts.adoc
Comment (nkakkar81) [reply modules/op-pipelines-concepts.adoc]: @siamaksade ^^

**Comment by nkakkar81** [comment]
@bmcelvee - Request you to please review this PR. The preview is still available. I will comment out the _topics.yml  file once you have reviewed and we are ready to merge. 
Thanks!!

**Review by vdemeester — APPROVED** [review]

File: modules/op-pipelines-concepts.adoc
Comment (bmcelvee) [inline modules/op-pipelines-concepts.adoc:8]: I would suggest s/:/. for this statement since it's not followed by a list, but a series of subheaders.

File: modules/op-pipelines-concepts.adoc
Comment (bmcelvee) [inline modules/op-pipelines-concepts.adoc:12]: Resources is capitalized inconsistently in this paragraph. Is it intentional based on the objects?

File: modules/op-pipelines-concepts.adoc
Comment (bmcelvee) [inline modules/op-pipelines-concepts.adoc:16]: s/auto-created/automatically created

File: modules/op-pipelines-concepts.adoc
Comment (bmcelvee) [inline modules/op-pipelines-concepts.adoc:24]: Just want to verify that the image goes through graphics for a style check. I asked a question about this in the team channel to make sure it's still standard practice.

File: modules/op-pipelines-concepts.adoc
Comment (bmcelvee) [inline modules/op-pipelines-concepts.adoc:22]: I would suggest s/else/but (or similar) - it might trigger confusion in the translation process.

File: openshift_pipelines/understanding-openshift-pipelines.adoc
Comment (bmcelvee) [inline openshift_pipelines/understanding-openshift-pipelines.adoc:17]: s/microservice based/microservice-based

File: openshift_pipelines/understanding-openshift-pipelines.adoc
Comment (bmcelvee) [inline openshift_pipelines/understanding-openshift-pipelines.adoc:24]: Please remove links to GitHub repos or request approval from Eric Rich.

File: modules/op-pipelines-concepts.adoc
Comment (nkakkar81) [reply modules/op-pipelines-concepts.adoc:12]: Yes, PipelineResources is a CRD defined in OpenShift Pipelines. Since in this section we are introducing the concepts definitions, we will be using the objects/components names defined here as is throughout the document.

File: modules/op-pipelines-concepts.adoc
Comment (nkakkar81) [reply modules/op-pipelines-concepts.adoc:24]: I have removed the image for this release. I will get this image checked by the graphics team and may be add it in a later release

File: openshift_pipelines/understanding-openshift-pipelines.adoc
Comment (nkakkar81) [reply openshift_pipelines/understanding-openshift-pipelines.adoc:24]: I have removed one link. For rest of the  link in this doc, I have received approval from the @RickJWagner. I will forward you the email with list of links for which the approval is received.

**Comment by nkakkar81** [comment]
@bmcelvee - All the comments have been incorporated. Let me know if you more suggestions, I will comment out the _topics.yaml file if you think its good to merge. Thanks!

**Comment by bmcelvee** [comment]
Thanks so much, @nkakkar81! This looks great. Please squash the commits on this PR and once we look into the Github link (for this PR and PR20692) we'll be ready to merge. 

I went ahead and applied the labels too.

**Comment by nkakkar81** [comment]
> Thanks so much, @nkakkar81! This looks great. Please squash the commits on this PR and once we look into the Github link (for this PR and PR20692) we'll be ready to merge.
> 
> I went ahead and applied the labels too.

Hi @bmcelvee - As discussed on slack, I have commented the links, and have  also squashed the comments. Request you to pls merge this PR. Thanks!!

**Pull request merged**
